Web Applications (extra-web) Survey on 20210928

MingcongBai opened the pull request:
Topic Description
-----------------

An survey for Web application updates (in extra-web), update where available.

Package(s) Affected
-------------------

```
extra-web/axel
extra-web/baidupcs-go
extra-web/chromium
extra-web/claws-mail
extra-web/clawsker
extra-web/curl
extra-web/darkhttpd
extra-web/dino
extra-web/dropbox
extra-web/elinks
extra-web/gajim
extra-web/google-chrome
extra-web/httpd
extra-web/httpie
extra-web/hugo
extra-web/irssi
extra-web/liferea
extra-web/mldonkey
extra-web/nextcloud-client
extra-web/nginx
extra-web/openresty
extra-web/opera
extra-web/otter-browser
extra-web/owncloud-client
extra-web/palemoon
extra-web/pidgin
extra-web/purple-matrix
extra-web/qbittorrent
extra-web/qownnotes
extra-web/qtox
extra-web/quaternion
extra-web/quiterss
extra-web/qutebrowser
extra-web/rclone
extra-web/rtv
extra-web/seamonkey
extra-web/squid
extra-web/srain
extra-web/sylpheed
extra-web/tg
extra-web/thunderbird
extra-web/vivaldi
extra-web/weechat
extra-web/znc
```

Security Update?
----------------

Yes

Build Order
-----------

See list above.

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

MingcongBai committed:
axel: update to 2.17.10

MingcongBai committed:
curl: update to 7.79.1; fix #3457

MingcongBai committed:
chromium: update to 94.0.4606.61; fix #3489

MingcongBai committed:
gajim: update to 1.3.2

MingcongBai committed:
dino: update to 0.2.2

MingcongBai committed:
darkhttpd: update to 1.13

MingcongBai committed:
irssi: update to 1.2.3

MingcongBai committed:
httpd: update to 2.4.49

MingcongBai committed:
httpie: update to 2.5.0

MingcongBai committed:
elinks: update to 0.12~pre6

MingcongBai committed:
clawsker: update to 1.3.4

MingcongBai committed:
mldonkey: update to 3.1.7+2

MingcongBai committed:
nextcloud-client: update to 3.3.4

MingcongBai committed:
google-chrome: update to 94.0.4606.61; fix #3489

MingcongBai committed:
dropbox: update to 131.4.3968

MingcongBai committed:
claws-mail: update to 4.0.0

MingcongBai committed:
baidupcs-go: update to 3.8.3

MingcongBai committed:
liferea: update to 1.13.6

MingcongBai committed:
hugo: update to 0.88.1

MingcongBai committed:
rclone: update to 1.56.1

MingcongBai committed:
opera: update to 79.0.4143.72; fix #3489

MingcongBai committed:
squid: update to 5.1

MingcongBai committed:
tg: update to 20200106

MingcongBai committed:
pidgin: update to 2.14.7

MingcongBai committed:
srain: update to 1.3.0

MingcongBai committed:
purple-matrix: update to 0.0.0+git20210626

MingcongBai committed:
openresty: update to 1.19.9.1

MingcongBai committed:
nginx: update to 1.21.3

MingcongBai committed:
qownnotes: update to 21.9.12

MingcongBai committed:
qbittorrent: update to 4.3.8

MingcongBai committed:
sylpheed: update to 3.7.0

MingcongBai committed:
otter-browser: update to 1.0.02

MingcongBai committed:
qtox: update to 1.17.3

MingcongBai committed:
quaternion: update to 0.0.95

MingcongBai committed:
palemoon: update to 29.4.1; fix #3491

MingcongBai committed:
rtv: update to 1.27.0

MingcongBai committed:
owncloud-client: update to 2.9.0.5150

MingcongBai committed:
seamonkey: update to 2.53.9.1; fix #3492

MingcongBai committed:
quiterss: update to 0.19.4

MingcongBai committed:
qutebrowser: update to 2.3.1

MingcongBai committed:
vivaldi: update to 4.2.2406.52; fix #3489

MingcongBai committed:
znc: update to 1.8.2

MingcongBai committed:
weechat: update to 3.3

MingcongBai committed:
thunderbird: update to 91.1.2

KexyBiscuit commented on extra-web/dropbox/autobuild/build:
```diff
@@ -1,11 +1,16 @@
     -type f -exec chmod -v 644 {} \;
-chmod -v 755 "$PKGDIR"/usr/lib/dropbox/dropboxd
-chmod -v 755 "$PKGDIR"/usr/lib/dropbox/dropbox
+chmod -v +x "$PKGDIR"/usr/lib/dropbox/*.so.*
```
```suggestion
chmod -v +x "$PKGDIR"/usr/lib/dropbox/*.so*
```

KexyBiscuit commented on extra-web/opera/autobuild/build:
```diff
@@ -1,9 +1,16 @@
 chmod -v 4755 "$PKGDIR"/usr/lib/opera/opera_sandbox
+
+abinfo "Setting executable bits for shared objects ..."
+chmod -v +x "$PKGDIR"/usr/lib/opera/*.so.*
```
```suggestion
chmod -v +x "$PKGDIR"/usr/lib/opera/*.so*
```

KexyBiscuit commented on extra-web/vivaldi/spec:
```diff
@@ -1,10 +1,10 @@
+VER=4.2.2406.52
 
 SRCS__ARM64="tbl::https://downloads.vivaldi.com/stable/vivaldi-stable_${VER}-1_arm64.deb"
 CHKSUMS__ARM64="sha384::cf85176e9b99e2ee9edded6f711b50a8f627dbe7933890dbc6e73b0fa3c98b046a6483d190b5d0f0af40fc06b0bc1935"
```
```suggestion
CHKSUMS__ARM64="sha384::555df414237d2df64b73aa32eb2a96bcf28646280c8fea4f7f154fa38b8b9f252ae8725acc06a88f3368e7c20ef2d8b5"
```

KexyBiscuit reviewed (changes requested):
A few changes.

KexyBiscuit commented:
/build chromium firefox

MingcongBai committed:
chromium: (Arch Linux patches) fix build

- Use Python 3 as Mojo interpreter.
- Also add a patch to fix Blink renderer compilation.
- Drop Debian desktop GL recognition patch, which breaks the build.

KexyBiscuit commented:
/build

KexyBiscuit committed:
chromium: remove patch which requires LLVM 13

KexyBiscuit commented:
/build chromium

KexyBiscuit committed:
mldonkey: upstream dropped the patches

KexyBiscuit commented:
/build mldonkey

KexyBiscuit committed:
quiterss: no debug symbols available

KexyBiscuit committed:
qbittorrent: no debug symbols available

KexyBiscuit commented:
/build quiterss qbittorrent

KexyBiscuit committed:
chromium: remove patch only has comment

KexyBiscuit committed:
Revert "mldonkey: upstream dropped the patches"

This reverts commit 8c76dde4d93e7206f74a8f14bb53b95017b51470.

KexyBiscuit committed:
mldonkey: update ocaml deps

KexyBiscuit commented:
/build mldonkey

KexyBiscuit commented:
`srain` FTBFS; SrainApp/srain#340

KexyBiscuit committed:
mldonkey: make patches dir whatever

KexyBiscuit commented:
/build mldonkey

KexyBiscuit committed:
mldonkey: mkdir uses p as forced

KexyBiscuit commented:
/build mldonkey

KexyBiscuit commented:
`firefox` and `seamonkey` won't build on GitHub Actions due to known issues; `mldonkey` still FTBFS; other build failures are fixed or unrelated.

MingcongBai committed:
chromium: update to 94.0.4606.71; fix #3489

MingcongBai committed:
mldonkey: fix build routine; lint scripts

MingcongBai committed:
nextcloud-client: fix CMAKE_DEF => CMAKE_AFTER

MingcongBai committed:
qbittorrent: disable build-time stripping

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
quiterss: disable build-time stripping

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
httpd: update to 2.4.50; fix #3530

MingcongBai committed:
opera: add executable bits to *.so*

Co-authored-by: Kexy Biscuit <kexybiscuit@aosc.io>

MingcongBai committed:
dropbox: add executable bits to *.so*

Co-authored-by: Kexy Biscuit <kexybiscuit@aosc.io>

MingcongBai committed:
vivaldi: update checksum for AArch64 binaries

Co-authored-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
qbittorrent: split dbg symbols again

KexyBiscuit committed:
quiterss: split dbg symbols again

KexyBiscuit reviewed (dismissed):
LGTM.

KexyBiscuit commented:
/build qbittorrent

KexyBiscuit commented:
/build srain

KexyBiscuit committed:
chromium: update to 94.0.4606.81; fix #3489

KexyBiscuit committed:
google-chrome: update to 94.0.4606.81; fix #3489

KexyBiscuit committed:
vivaldi: update to 4.3.2439.44; fix #3489

KexyBiscuit commented:
/build chromium google-chrome vivaldi

KexyBiscuit committed:
chromium: restore incorrectly dropped patch

Revert "chromium: remove patch which requires LLVM 13"

This reverts commit 46bc9f60f1b114dc6013cffb859a239a588399a0.

KexyBiscuit commented:
/build chromium

KexyBiscuit commented:
/build chromium

KexyBiscuit committed:
chromium: adapt patch for 94.0.4606.81

KexyBiscuit commented:
/build chromium

KexyBiscuit committed:
chromium: remove trailing spaces in patch

KexyBiscuit commented:
/build chromium

MingcongBai committed:
chromium: use system eu-strip for build time

MingcongBai committed:
httpd: update to 2.4.51; fix #3530

MingcongBai committed:
mldonkey: limit to amd64

MingcongBai committed:
vivaldi: update checksum for arm64

MingcongBai committed:
chromium: limit to amd64, arm64

MingcongBai committed:
seamonkey: limit to amd64, arm64

- SeaMonkey introduced Chromium components, hampering platform compatibility.

KexyBiscuit commented on extra-web/chromium/autobuild/prepare:
```diff
@@ -46,3 +46,8 @@ ln -sv /usr/bin/node \
 ln -sv /usr/bin/java \
     "$SRCDIR"/third_party/jdk/current/bin/
+
+abinfo "Fedora: Use system eu-strip ..."
```
I'd prefer add link to the patch... NVM.

KexyBiscuit reviewed (approved):
LGTM overall.
